geoviews/util.py

```diff
@@ -677,13 +685,21 @@ def from_xarray(da, crs=None, apply_transform=False, nan_nodata=False, **kwargs)
     if crs:
         kwargs['crs'] = crs
     elif hasattr(da, 'crs'):
+        # xarray.open_rasterio (not supported since April 2023)
         try:
             kwargs['crs'] = process_crs(da.crs)
         except Exception:
-            param.main.warning('Could not decode projection from crs string %r, '
-                               'defaulting to non-geographic element.' % da.crs)
+            warn(f'Could not decode projection from crs string {da.crs}, '
+                  'defaulting to non-geographic element.')
+    elif hasattr(da, 'rio') and da.rio.crs is not None:
+        # rioxarray.open_rasterio
+        try:
+            kwargs['crs'] = process_crs(da.rio.crs.to_proj4())
+        except Exception:
+            warn(f'Could not decode projection from crs string {da.rio.crs}, '
+                  'defaulting to non-geographic element.')
 
-    coords = list(da.coords)
+    coords = list(da.coords.dims)
```

Comment:
```suggestion
    coords = list(da.dims)
```

Reply:
I'm not that familiar with xarray to know what the difference is between `da.coords.dims` and `da.dims`. Do you have a link to a quick explanation?

Reply:
I think they access the same thing; da.dims is just the shorthand (and I think more standard).

```
import xarray as xr
ds = xr.tutorial.open_dataset("air_temperature")
ds.coords["tau"] = ds.coords["time"] - ds.coords["time"][0]
print("DIMS", ds.dims)
print("COORDS", ds.coords)
print("DIMS", ds.coords.dims)
```

The ones prefixed with * are dimensions (dims)
<img width="746" alt="image" src="https://github.com/holoviz/geoviews/assets/15331990/e0d482ef-bf22-4031-8dd8-b7217b327dbb">

Reply:
You are right. 

![image](https://github.com/holoviz/geoviews/assets/19758978/1555e377-88ae-4e42-a58d-23a2ae7dce52)